tests/utils/base_test_case.py

```diff
@@ -28,44 +28,131 @@
+                                'response-content-type=application%2Foctet-stream&'\
+                                'AWSAccessKeyId=AKIAIOSFODNN7EXAMPLE'\
+                                '&Signature=VLCdYUh8skB6cRqo7RUfGrycsKo%3D&Expires=1573768889'
 
     def setUp(self):
         self._remove_file(self.test_captured_output_file)
 
-    def _setup_mock_response(self, mock_server, request, response, with_auth=True):
-        self._remove_keys(request, ['params', 'json'])
-        if with_auth:
-            self._add_auth(request)
-        mock_server.request(text=response, **request)
+    def _setup_mock_api_response(self, mock_server, request, response):
+        request_parameters = self.convert_polyswarm_api_request_to_mock_request_parameters(request)
+        mock_server.request(text=response, **request_parameters)
 
-    def _create_search_hash_request(self, file_hash, with_instances=True, with_metadata=True):
-        request = self.request_generator.search_hash(resources.Hash.from_hashable(file_hash), with_instances, with_metadata)
-        self.convert_polyswarm_api_request_to_mock_request(request)
+    @staticmethod
+    def _setup_mock_url_response(mock_server, url, response):
+        request_parameters = {'method': 'GET',
+                              'url': url}
+        mock_server.request(body=response, **request_parameters)
+
+    def _setup_mock_api_download(self, mock_server, request, download_file):
+        request_parameters = self.convert_polyswarm_api_request_to_mock_request_parameters(request)
+        mock_server.request(body=download_file, **request_parameters)
+
+    def _create_search_hash_request(self, hash_value, offset=None, limit=None):
+        request = self.request_generator.search_hash(resources.Hash.from_hashable(hash_value))
+        self._add_pagination_params(request, offset, limit)
+        return request
+
+    def _create_search_metadata_request(self, query, offset=None, limit=None):
+        request = self.request_generator.search_metadata(resources.MetadataQuery(query))
+        self._add_pagination_params(request, offset, limit)
         return request
 
-    def _create_search_metadata_request(self, query, with_instances=True, with_metadata=True):
-        request = self.request_generator.search_metadata(resources.MetadataQuery(query), with_instances, with_metadata)
-        self.convert_polyswarm_api_request_to_mock_request(request)
+    def _create_scan_submission_lookup_request(self, submission_uuid):
+        request = self.request_generator.lookup_uuid(submission_uuid)
         return request
 
-    def _create_hunt_live_list_request(self):
+    def _create_scan_submission_submit_request(self, artifact):
+        request = self.request_generator.submit(resources.LocalArtifact(artifact))
+        return request
+
+    def _create_scan_submission_rescan_request(self, hash_value, hash_type='sha256'):
+        request = self.request_generator.rescan(resources.Hash(hash_value, hash_type))
```

Comment:
Request generation should be in `polyswarm-api` tests, not `cli`. These are internal methods and classes of `api` and are out of this scope. Things would be simpler if these were in `api` tests, as we have to replicate methods to mock things that should not be here.

Reply:
These are only test helper methods meant to make tests cleaner. They are required if we want to do integration tests (they are only used by integration tests, thus it seems odd to me to move them to `polyswarm-api`).

Reply:
Ok, we are only testing some cases here, fair enough. :+1: